app/src/features/reports/CompareMapNavReport.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useEffect } from "react";
+import { Grid } from "@material-ui/core";
+import UsaChloroplethMap from "../charts/UsaChloroplethMap";
+import TableChart from "../charts/TableChart";
+import WithDatasets from "../../utils/WithDatasets";
+import useDatasetStore from "../../utils/useDatasetStore";
+import variableProviders from "../../utils/variableProviders";
+import { Breakdowns } from "../../utils/Breakdowns";
+import VariableProvider from "../../utils/variables/VariableProvider";
+import { STATE_FIPS_MAP } from "../../utils/Fips";
+import Alert from "@material-ui/lab/Alert";
+import Breadcrumbs from "@material-ui/core/Breadcrumbs";
+import Link from "@material-ui/core/Link";
+import Typography from "@material-ui/core/Typography";
+
+interface Geo {
+  fips: number;
+  name: string;
+}
+
+function Map(props: {
+  fipsGeo: number;
+  data: Record<string, any>[];
+  updateGeoCallback: Function;
+}) {
+  function getStateGeoFromPropFips() {
+    return props.fipsGeo === 0
+      ? undefined
+      : { fips: props.fipsGeo, name: STATE_FIPS_MAP[props.fipsGeo] };
+  }
+
+  const [state, setState] = useState<Geo | undefined>(
+    getStateGeoFromPropFips()
+  );
+  const [county, setCounty] = useState<Geo | undefined>();
+
+  useEffect(() => {
+    props.updateGeoCallback(state ? state.fips : 0);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [state]);
+
+  useEffect(() => {
+    setState(getStateGeoFromPropFips());
+    setCounty(undefined);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [props.fipsGeo]);
```

Comment:
I think this is a bit of an anti-pattern in React, where the the state geography selection is partially controlled by this component and partially controlled by the parent component. It can lead to issues with the state getting out of sync or incorrectly overwritten. 

I'd recommend having the state geography entirely controlled via a prop that's passed in, and remove the state from this class, and this class just uses the callback to tell the parent to change the selection. This also allows you to get rid of the useEffect calls, which can be brittle since the state won't update in the same react update

Reply:
ya good call!